Library/OcCryptoLib/Streebog.c

```diff
@@ -176,201 +176,203 @@ Uint512uCpy (
 STATIC
 VOID
 Stage2 (
-  StreebogContext  *CTX,
-  CONST UINT8      *data
+  STREEBOG_CONTEXT  *Context,
+  CONST UINT8       *Data
   )
 {
   union uint512_u  m;
 
-  MasCpy ((UINT8 *)&m, data);
-  g (&(CTX->h), &(CTX->N), (CONST UINT8 *)&m);
+  MasCpy ((UINT8 *)&m, Data);
+  g (&(Context->h), &(Context->N), (CONST UINT8 *)&m);
 
-  Add512 (&(CTX->N), &buffer512, &(CTX->N));
-  Add512 (&(CTX->Sigma), &m, &(CTX->Sigma));
+  Add512 (&(Context->N), &buffer512, &(Context->N));
+  Add512 (&(Context->Sigma), &m, &(Context->Sigma));
 }
 
 STATIC
 VOID
 Stage3 (
-  StreebogContext  *CTX
+  STREEBOG_CONTEXT  *Context
   )
 {
   union uint512_u  buf = {
     { 0 }
   };
 
- #if defined (__BYTE_ORDER__) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
-  buf.QWORD[0] = CTX->bufsize << 3;
- #elif defined (__BYTE_ORDER__) && __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
-  buf.QWORD[0] = BSWAP64 (CTX->bufsize << 3);
- #else
+#if (defined (__BYTE_ORDER__) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__) || defined(MDE_CPU_IA32) || defined(MDE_CPU_X64)
```

Comment:
Please introduce separate macros STREEBOG_LITTLE_ENDIAN and STREEBOG_BIG_ENDIAN in Streebog.h to reduce the length.